### ISSUE TITLE
Why is Kday.kday_before(~D[2023-09-30), 5) Sep 22?

### ISSUE THREAD

**Issue by puruzio** [kipcole9/kday#2]
9/30 is Saturday, so I expected the above expression to return Sep 29 (Friday). Thanks.

**Comment by kipcole9** [comment]
Ouch, that's not good. Thanks for the report @puruzio.  I will fix this over the weekend.  For now you can use `Kday.kday_on_or_before/2` that does return the correct value:
```elixir
# Incorrect
iex> Kday.kday_before(~D[2023-09-30], 5)
~D[2023-09-22]

# Correct
iex> Kday.kday_on_or_before(~D[2023-09-30], 5)
~D[2023-09-29]
```

**Comment by puruzio** [comment]
Thanks for your super-fast response! Btw,  kday_on_or_before works for `kday_before(~D[2023-09-30], 5)` but shouldn't the following return 9/30 instead of 9/23?
iex(172)> Kday.kday_on_or_before(~D[2023-09-30], 6) 
~D[2023-09-23]

**Comment by kipcole9** [comment]
Yes, you are right. The bug is in converting iso days (internal integer representation of a date) into a day of the week. And it's horrible - all of the previous tests work (and I just hand verified them) but clearly there is a bug.

I'm on it! (but realistically won't get closed until the weekend).

**Comment by puruzio** [comment]
No rush. Thank you!!

**Comment by kipcole9** [comment]
I've push a commit that I think fixes the issue but I want to create more tests to be sure. Would you consider configuring the library from GitHub and giving it a try?  

Please reopen if the commit doesn't fix it of course!

**Comment by puruzio** [comment]
Looks good. Thank you!!

**Comment by kipcole9** [comment]
I've published [Kday version 1.0.2](https://hex.pm/packages/kday/1.0.2) with the following changelog entry:

### Bug Fixes

* Fix `Kday.iso_days_to_day_of_week/1`. Thanks to @puruzio for the report. Closes #2.

Thanks for the report (and thanks for finding value with the library!).
